internal/gengapic/testdata/foo_example.want

```diff
@@ -8,7 +8,7 @@ func ExampleNewFooClient() {
 	_ = c
 }
 
-func ExampleFooClient_GetEmptyThing() {
+func ExampleFooGrpcClient_GetEmptyThing() {
```

Comment:
I don't think we should be changing these. These are actually documentation examples, and the naming is important. Since the "public interface" is `FooClient`, the examples should correlate to `FooClient_GetEmptyThing`.

I think you can revert the example changes altogether for now since there is no transport-specific different, and because the gRPC client is internal.

Reply:
I was torn about this change in particular; eventually, the examples and documentation should clearly and unambiguously reflect the underlying transport being used. Having a convention of "the user visible default is gGRPC, anything with Rest in the name is Rest" is less than ideal.